src/scorer.py

```diff
@@ -16,16 +15,19 @@ class Scorer:
     PENALTY_INSERTION = 100
     PENALTY_DELETION = 100
 
-    def __init__(self, target_o: str, *, stack_differences: bool, debug_mode: bool):
+    def __init__(
+        self,
+        target_o: str,
+        *,
+        stack_differences: bool,
+        algorithm: str,
+        debug_mode: bool,
+    ):
         self.target_o = target_o
         self.arch = get_arch(target_o)
         self.stack_differences = stack_differences
+        self.algorithm = algorithm
         self.debug_mode = debug_mode
-        _, self.target_seq = self._objdump(target_o)
-        self.differ: difflib.SequenceMatcher[str] = difflib.SequenceMatcher(
-            autojunk=False
-        )
-        self.differ.set_seq2([line.mnemonic for line in self.target_seq])
```

Comment:
It would be good if we could keep this differ around in the difflib case -- the docs say
```
set_seq2(b) method of difflib.SequenceMatcher instance
    Set the second sequence to be compared.
...    
    SequenceMatcher computes and caches detailed information about the
    second sequence, so if you want to compare one sequence S against
    many sequences, use .set_seq2(S) once and call .set_seq1(x)
    repeatedly for each of the other sequences.
```
and this feels like a performance win we potentially care about